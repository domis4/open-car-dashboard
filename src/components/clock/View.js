import React from 'react'
import Typography from '@material-ui/core/Typography'

const View = ({ classes, time }) => (
  <Typography component="p" className={classes.text}>
    {time}
  </Typography>
)

export default View
