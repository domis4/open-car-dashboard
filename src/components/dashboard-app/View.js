//@flow
import React from 'react'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import placeholder from '../../assets/placeholder/placeholder-app-icon.png'
import Zoom from '@material-ui/core/Zoom'

type propTypes = {
  classes: {},
  image?: any,
  title?: string,
  callback?: () => {},
}

const View = ({ classes, image, title, callback = () => {} }: propTypes) => (
  <Zoom in={true}>
    <Card className={classes.card} onClick={() => callback()}>
      <CardActionArea>
        <CardMedia component="img" className={classes.media} image={image || placeholder} />
        <CardContent>
          <Typography gutterBottom variant="headline" component="h2" className={classes.text}>
            {title || 'App Title'}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  </Zoom>
)

export default View
