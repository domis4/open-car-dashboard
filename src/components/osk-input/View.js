import React from 'react'
import Keyboard, { KeyboardButton } from 'react-screen-keyboard'

const View = ({value, }) => (
  <div>
    <input
      onInput={this.handleInput}
      value={value}
      onFocus={this.handleFocus}
      ref={input => {
        this.input = input
      }}
    />
    <Keyboard
      inputNode={inputNode}
      rightButtons={[
        <ClickOnKeyPressWrap key="enter">
          <KeyboardButton onClick={this.handleLoginUser} value="asdf" classes="keyboard-submit-button" />
        </ClickOnKeyPressWrap>,
      ]}
    />
  </div>
)

export default View
