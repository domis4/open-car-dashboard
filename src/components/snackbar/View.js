//@flow
import React from 'react'
import Snackbar from '@material-ui/core/Snackbar'

type ViewPropTypes = {
  text: string,
  close: () => {},
}

const View = ({ text, close }: ViewPropTypes) => (
  <Snackbar
    anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
    open={true}
    onClose={close}
    message={<span>{text}</span>}
  />
)

export default View
