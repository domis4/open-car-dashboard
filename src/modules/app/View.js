import React from 'react'
import { Provider } from 'react-redux'
import { store, history } from '../../create-store'
import { compose } from 'recompose'
import { Route, Switch } from 'react-router'
import { ConnectedRouter } from 'connected-react-router'
import { MuiThemeProvider, withStyles, createMuiTheme } from '@material-ui/core/styles'
import Fade from '@material-ui/core/Fade'
import styles from './styles'
import Bootlogo from '../bootlogo'
import Dashboard from '../dashboard'
import MainMenu from '../main-menu'
import MusicPlayer from '../music-player'
import Settings from '../settings'
import Maps from '../google-maps'
import Notifications from '../notifications'

function getTheme(theme) {
  return createMuiTheme({
    overrides: {
      MuiPaper: {
        root: {
          backgroundColor: '#000',
        },
      },
      MuiSvgIcon: {
        root: {
          color: '#fff',
        },
      },
    },
    palette: {
      type: theme.paletteType,
      background: {
        default: theme.paletteType === 'light' ? '#000' : '#fff',
      },
    },
  })
}

const theme = getTheme({
  paletteType: 'dark',
})

const App = ({ classes, isBootIconVisible, setBootIconVisible }) => (
  <Provider {...{ store }}>
    <ConnectedRouter history={history}>
      <MuiThemeProvider theme={theme}>
        <Bootlogo callback={() => setBootIconVisible(false)} />
        {!isBootIconVisible && (
          <Fade {...{ in: !isBootIconVisible, timeout: 600, className: classes.root }}>
            <div className={classes.root}>
              <div className={classes.app}>
                <MainMenu history={history} />
                <main className={classes.content}>
                  <Notifications />
                  <Switch className={classes.router}>
                    <Route exact path="/" component={Dashboard} history={history} />
                    <Route path="/music/:tab?" component={MusicPlayer} history={history} />
                    <Route path="/settings" component={Settings} history={history} />
                    <Route path="/maps" component={Maps} history={history} />
                  </Switch>
                </main>
              </div>
            </div>
          </Fade>
        )}
      </MuiThemeProvider>
    </ConnectedRouter>
  </Provider>
)

export default compose(withStyles(styles))(App)
