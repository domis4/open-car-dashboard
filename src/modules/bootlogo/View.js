import React from 'react'
import Logo from '../../assets/bootlogo/vw.svg'
import SVGInline from 'react-svg-inline'
import Fade from '@material-ui/core/Fade'
import Typography from '@material-ui/core/Typography'

const View = ({ showElement, classes, fadeIn }) => (
  <Fade {...{ in: fadeIn, timeout: 600, className: classes.root }}>
    <div {...{ className: classes.root }}>
      {showElement && (
        <div {...{ className: classes.inner }}>
          <SVGInline {...{ svg: Logo }} />
          <Typography component="h1" className={classes.text}>
            Welcome
          </Typography>
        </div>
      )}
    </div>
  </Fade>
)

export default View
