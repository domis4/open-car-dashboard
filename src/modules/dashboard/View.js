//@flow
import React from 'react'
import Paper from '@material-ui/core/Paper'
import Grid from '@material-ui/core/Grid'
import DashboardApp from '../../components/dashboard-app'
import musicImage from '../music-player/icon.png'
import settingsIcon from '../settings/icon.png'
import googleMapsIcon from '../google-maps/icon.png'

type propTypes = {
  classes: {},
  history: {},
}

const View = ({ classes, history }: propTypes) => (
  <div className={classes.root}>
    <Grid container spacing={24}>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp image={musicImage} title="Music" callback={() => history.push('/music')} />
        </Paper>
      </Grid>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp image={settingsIcon} title="Settings" callback={() => history.push('/settings')} />
        </Paper>
      </Grid>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp image={googleMapsIcon} title="Google Maps" callback={() => history.push('/maps')} />
        </Paper>
      </Grid>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp />
        </Paper>
      </Grid>
    </Grid>
  </div>
)

export default View
