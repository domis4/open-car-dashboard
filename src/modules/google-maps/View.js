//@flow
import React from 'react'
import Zoom from '@material-ui/core/Zoom'

type propTypes = {
  classes: {},
  tabValue?: string,
  setTabValue: string => void,
}

const View = ({ classes, tabValue = 'lists', setTabValue }: propTypes) => (
  <Zoom in={true}>
    <iframe className={classes.iframe} src="http://maps.openrouteservice.org" scrolling="no" />
  </Zoom>
)

export default View
