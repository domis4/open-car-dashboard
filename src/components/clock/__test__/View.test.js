import React from 'react'
import { shallow } from 'enzyme'
import View from '../View'

describe('clock component View.js', () => {
  it('expects component to match snapshot', () => {
    const wrapper = shallow(<View {...{ classes: {}, time: '16:10' }} />).dive()

    expect(wrapper).toMatchSnapshot()
  })
  it('expects component to render a Typography component', () => {
    const wrapper = shallow(<View {...{ classes: {}, time: '16:10' }} />).dive()

    expect(wrapper.find('Typography')).toHaveLength(1)
  })
  it('expects component to render a paragraph element containing the curren time', () => {
    const wrapper = shallow(<View {...{ classes: {}, time: '16:10' }} />).dive()
    const elem = '<p class="MuiTypography-root-01 MuiTypography-body1-010">16:10</p>'

    expect(wrapper.html()).toEqual(elem)
  })
})
