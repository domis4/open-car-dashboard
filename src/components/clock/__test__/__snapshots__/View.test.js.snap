// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clock component View.js expects component to match snapshot 1`] = `
<Typography
  align="inherit"
  classes={
    Object {
      "alignCenter": "MuiTypography-alignCenter-014",
      "alignJustify": "MuiTypography-alignJustify-016",
      "alignLeft": "MuiTypography-alignLeft-013",
      "alignRight": "MuiTypography-alignRight-015",
      "body1": "MuiTypography-body1-010",
      "body2": "MuiTypography-body2-09",
      "button": "MuiTypography-button-012",
      "caption": "MuiTypography-caption-011",
      "colorError": "MuiTypography-colorError-025",
      "colorInherit": "MuiTypography-colorInherit-020",
      "colorPrimary": "MuiTypography-colorPrimary-021",
      "colorSecondary": "MuiTypography-colorSecondary-022",
      "colorTextPrimary": "MuiTypography-colorTextPrimary-023",
      "colorTextSecondary": "MuiTypography-colorTextSecondary-024",
      "display1": "MuiTypography-display1-05",
      "display2": "MuiTypography-display2-04",
      "display3": "MuiTypography-display3-03",
      "display4": "MuiTypography-display4-02",
      "gutterBottom": "MuiTypography-gutterBottom-018",
      "headline": "MuiTypography-headline-06",
      "noWrap": "MuiTypography-noWrap-017",
      "paragraph": "MuiTypography-paragraph-019",
      "root": "MuiTypography-root-01",
      "subheading": "MuiTypography-subheading-08",
      "title": "MuiTypography-title-07",
    }
  }
  color="default"
  component="p"
  gutterBottom={false}
  headlineMapping={
    Object {
      "body1": "p",
      "body2": "aside",
      "display1": "h1",
      "display2": "h1",
      "display3": "h1",
      "display4": "h1",
      "headline": "h1",
      "subheading": "h3",
      "title": "h2",
    }
  }
  noWrap={false}
  paragraph={false}
  variant="body1"
>
  16:10
</Typography>
`;
